lib/LaTeXML/Post.pm

```diff
@@ -447,7 +447,7 @@ sub convertXMTextContent {
   foreach my $node (@nodes) {
     if ($node->nodeType == XML_TEXT_NODE) {
       my $string = $node->textContent;
-      $string =~ s/^\s+/$NBSP/; $string =~ s/\s+$/$NBSP/;    # should we???
```

Comment:
Hmm... did I ever follow up to this? I did end up making some rather deeper patches to LaTeXML to define our own serializer so that we get the aesthetic & debugging benefit of indented XML, but done correctly respecting the document schema.  In otherwords, now ignorable whitespace *is* ignorable.   So, now when I'm dealing with whitespace in XMText/mtext/etc, I *know* that the whitespace is real.  I've also adapted some of the related methods for converting whitespace, optionally, to non-break whitespace.  It does the conversion for mathml, but not openmath, as seems to be the request here.

So, you guys are going to want to do a merge from the latexml base, but expect some conflicts.